analyzers/src/SonarAnalyzer.CFG/LiveVariableAnalysis/RoslynLiveVariableAnalysis.cs

```diff
@@ -70,10 +72,48 @@ public ISymbol ParameterOrLocalSymbol(IOperation operation)
     protected override State ProcessBlock(BasicBlock block)
     {
         var ret = new RoslynState(this);
-        ret.ProcessBlock(Cfg, block);
+        ret.ProcessBlock(Cfg, block, flowCaptureOperations);
         return ret;
     }
 
+    private void ResolveCaptures(ControlFlowGraph cfg)
+    {
+        foreach (var operation in cfg.Blocks.Where(x => x.IsEnclosedIn(ControlFlowRegionKind.LocalLifetime)).SelectMany(x => x.Operations).ToExecutionOrder())
```

Comment:
You could apply those SC suggestion here (having multine foreach is fine). It will be especially easier with the `As...` extension, followed by `.WhereNotNull()`